components/motor/gpio/loop.go

```diff
@@ -38,149 +23,80 @@ func (m *EncodedMotor) State(ctx context.Context) ([]float64, error) {
 func (m *EncodedMotor) updateControlBlock(ctx context.Context, setPoint, maxVel float64) error {
 	// Update the Trapezoidal Velocity Profile block with the given maxVel for velocity control
 	velConf := control.BlockConfig{
-		Name: m.blockNames[blockNameTrapezoidal],
+		Name: m.blockNames["trapezoidalVelocityProfile"][0],
 		Type: "trapezoidalVelocityProfile",
 		Attribute: rdkutils.AttributeMap{
 			"max_vel":    maxVel,
 			"max_acc":    30000.0,
 			"pos_window": 0.0,
 			"kpp_gain":   0.45,
 		},
-		DependsOn: []string{m.blockNames[blockNameConstant], m.blockNames[blockNameEndpoint]},
+		DependsOn: []string{m.blockNames["constant"][0], m.blockNames["endpoint"][0]},
 	}
-	if err := m.loop.SetConfigAt(ctx, m.blockNames[blockNameTrapezoidal], velConf); err != nil {
+	if err := m.loop.SetConfigAt(ctx, m.blockNames["trapezoidalVelocityProfile"][0], velConf); err != nil {
 		return err
 	}
 
 	// Update the Constant block with the given setPoint for position control
 	posConf := control.BlockConfig{
-		Name: m.blockNames[blockNameConstant],
+		Name: m.blockNames["trapezoidalVelocityProfile"][0],
 		Type: "constant",
 		Attribute: rdkutils.AttributeMap{
 			"constant_val": setPoint,
 		},
 		DependsOn: []string{},
 	}
```

Comment:
ah yeah this is related to my comment in the [base pr](https://github.com/viamrobotics/rdk/pull/3615#discussion_r1504771889), but I was thinking we could have a constructor function in the `controls` package to build a constant block config, since every implementer will have to make this struct. Up to you if you think user's need help making this config, and also fine to leave as is